pkg/services/comments/commentmodel/permissions.go

```diff
@@ -104,6 +105,12 @@ func (c *PermissionChecker) CheckWritePermissions(ctx context.Context, orgId int
 		if !c.features.IsEnabled(featuremgmt.FlagAnnotationComments) {
 			return false, nil
 		}
+		if c.features.IsEnabled(featuremgmt.FlagAccesscontrol) {
+			evaluator := accesscontrol.EvalPermission(accesscontrol.ActionAnnotationsWrite, accesscontrol.ScopeAnnotationsTypeDashboard)
```

Comment:
Quick question, what made you choose this over some SQL filtering like what you did for read?

Reply:
Purely because it seemed simpler to do it this way instead of extending annotation SQL filtering. But I am considering doing this in future if time allows (if we extend SQL filtering, I also want to refactor the way we do access checks in annotation handlers).